[BUG] `make types` doesn't specify `mypy` path

# Bug Details :warning::bug:

## Description

`make types` does not specify the path to `mypy`, and as such may run a version of `mypy` which is not associated with the virtual environment. This can lead to `mypy` errors for example if modules are not installed in the environment that mypy exists in.

## Steps to Reproduce

- [ ] Is the bug reproducible?

1. In a fresh shell in `across-server` directory, run `make types`.
2. See errors.

### Expected behavior

`mypy` runs without error.

### Actual behavior

`mypy` gives errors like this:
```python
mypy
across_server/auth/security.py:3: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
across_server/auth/security.py:4: error: Cannot find implementation or library stub for module named "fastapi.security"  [import-not-found]
across_server/core/exceptions.py:4: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
across_server/core/config.py:1: error: Cannot find implementation or library stub for module named "pydantic_settings"  [import-not-found]
across_server/util/decorators/local_only_route.py:4: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
migrations/versions/2024_10_25_1124-023e7f96f47f_init.py:12: error: Cannot find implementation or library stub for module named "alembic"  [import-not-found]
migrations/versions/2024_10_25_1124-023e7f96f47f_init.py:13: error: Cannot find implementation or library stub for module named "geoalchemy2"  [import-not-found]
across_server/db/config.py:1: error: Cannot find implementation or library stub for module named "pydantic_settings"  [import-not-found]
across_server/auth/tokens/base_token.py:6: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
across_server/auth/tokens/base_token.py:6: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
across_server/db/models.py:7: error: Cannot find implementation or library stub for module named "geoalchemy2"  [import-not-found]
migrations/env.py:4: error: Cannot find implementation or library stub for module named "alembic"  [import-not-found]
migrations/env.py:5: error: Cannot find implementation or library stub for module named "geoalchemy2"  [import-not-found]
across_server/routes/role/service.py:4: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
across_server/routes/group/service.py:4: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
across_server/routes/group/role/service.py:4: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
across_server/auth/service.py:4: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
across_server/auth/strategies.py:5: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
across_server/auth/strategies.py:6: error: Cannot find implementation or library stub for module named "fastapi.security"  [import-not-found]
across_server/auth/router.py:5: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
across_server/routes/user/service.py:4: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
across_server/routes/user/router.py:4: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
across_server/routes/role/router.py:4: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
across_server/routes/group/role/router.py:4: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
across_server/routes/group/router.py:4: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
migrations/seeds/footprints.py:1: error: Cannot find implementation or library stub for module named "geoalchemy2"  [import-not-found]
migrations/seeds/footprints.py:2: error: Skipping analyzing "shapely.geometry": module is installed, but missing library stubs or py.typed marker  [import-untyped]
across_server/main.py:1: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
tests/conftest.py:3: error: Cannot find implementation or library stub for module named "pytest_asyncio"  [import-not-found]
tests/conftest.py:4: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
tests/user/router_test.py:3: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
tests/user/router_test.py:5: error: Cannot find implementation or library stub for module named "pytest_asyncio"  [import-not-found]
tests/auth/router_test.py:4: error: Cannot find implementation or library stub for module named "fastapi"  [import-not-found]
tests/auth/router_test.py:6: error: Cannot find implementation or library stub for module named "pytest_asyncio"  [import-not-found]
Found 33 errors in 25 files (checked 78 source files)
make: *** [types] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -183,7 +183,7 @@ lint: ## Run linting
 	@$(VENV_BIN)/pre-commit run;
 
 types: ## Run type checks
-	mypy
+	@$(VENV_BIN)/mypy;
 
 
 # Group: Database

